core/src/main/java/org/elasticsearch/discovery/zen/PublishClusterStateAction.java

```diff
@@ -377,14 +382,16 @@ protected void handleIncomingClusterStateRequest(BytesTransportRequest request,
             in = request.bytes().streamInput();
         }
         in.setVersion(request.version());
+        // only custom cluster state named writeables are known here:
```

Comment:
I'd remove this note. We expect that only custom cluster state named writeables are used here but we support any named writeables. While we don't use this now we don't actively prevent using it and a plugin might use it.